controllers/authentication/openidconnect_controller.go

```diff
@@ -98,12 +108,10 @@ func (r *OpenIDConnectReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		opts.UsernamePrefix = config.Name + "/"
 	}
 
-	auth, err := oidc.NewForkedAuthenticator(oidc.OptionsForked{
-		CA:      config.Spec.CABundle,
-		Options: opts,
-	})
+	auth, err := oidc.New(opts)
```

Comment:
This is not sufficient. The `oidc.New(opts)` would run in a `go routine` if no keyset is provided - https://github.com/enj/kubernetes/blob/b5a1a45d48b4e90e54f512fc829b2ab9866b282e/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go#L315-L332

Reply:
agreed. I think this is related to https://github.com/gardener/oidc-webhook-authenticator/issues/2